zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryControllerTest.java

```diff
@@ -0,0 +1,393 @@
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(userTaskServices, never()).search(any(UserTaskQuery.class));
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithMissingSortField() {
+    // given
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "order": "asc"
+                }
+            ]
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Sort field must not be null",
+          "instance": "%s"
+        }""",
+            USER_TASKS_SEARCH_URL);
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(userTaskServices, never()).search(any(UserTaskQuery.class));
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithInvalidDate() {
+    // given
+    final var request =
+        """
+        {
+            "filter": {
+                "creationDate": {
+                    "from": "invalid-date"
+                }
+            }
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Invalid date format for 'from': Text 'invalid-date' could not be parsed at index 0",
+          "instance": "%s"
+        }""",
+            USER_TASKS_SEARCH_URL);
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(userTaskServices, never()).search(any(UserTaskQuery.class));
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithConflictingPagination() {
+    // given
+    final var request =
+        """
+        {
+            "page": {
+                "searchAfter": ["a"],
+                "searchBefore": ["b"]
+            }
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Error: Both searchAfter and searchBefore cannot be set at the same time.",
```

Comment:
❌ Heads-up: This will need to change after you rebased your PR since we aligned the problem response handling.
```suggestion
          "type": "about:blank",
          "title": "INVALID_ARGUMENT",
          "status": 400,
          "detail": "Error: Both searchAfter and searchBefore cannot be set at the same time.",
```